### PR TITLE
fix race condition on progress that could cause deadlock

### PR DESCRIPTION
**Pull request by moadibfr**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | yes
| 🚀 New feature?   | no
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | 
| ❓ Documentation  | no 

## Description
fix race condition on progress that could cause deadlock

### CODE REVIEW

File: pkg/output/progress_test.go [hunk 1]
```diff
@@ -7,13 +7,20 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func TestProgressTimeout(t *testing.T) {
+func TestProgressTimeoutDoesNotInc(t *testing.T) {
+	progress := NewProgress()
+	progress.Start()
+	progress.Stop() // should not hang
+	progress.Inc()  // should not hang
+	assert.Equal(t, progress.Val(), uint64(0))
```
Comment (sundowndev) [inline pkg/output/progress_test.go:15]: You mixed expected and actual value here

```diff
-assert.Equal(t, progress.Val(), uint64(0))
+assert.Equal(t, uint64(0), progress.Val())
```

File: pkg/output/progress_test.go [hunk 2]
```diff
@@ -7,13 +7,20 @@ import (
...
+}
+
+func TestProgressTimeoutDoesNotHang(t *testing.T) {
 	progress := NewProgress()
 	progress.Start()
 	time.Sleep(progressTimeout + 1)
 	progress.Inc()  // should not hang
 	progress.Stop() // should not hang
-	assert.Equal(t, uint64(0), progress.Val())
```
Comment (sundowndev) [inline pkg/output/progress_test.go:16]: Isn't it relevant to assert this value?

File: pkg/output/progress_test.go (hunk 1 above)
Comment (moadibfr) [reply pkg/output/progress_test.go:15]: :+1: fixed

File: pkg/output/progress_test.go (hunk 2 above)
Comment (moadibfr) [reply pkg/output/progress_test.go:16]: not realy but I changed the test to be sure the timeout was reached so I could put this test back

**Comment by sundowndev** [comment]
LGTM

**Review by sundowndev — APPROVED** [review]
LGTM

**Pull request merged**